mungegithub/mungers/testowner/owner_test.go

```diff
@@ -13,38 +17,82 @@ func TestCanonize(t *testing.T) {
 		"[k8s.io] blah {Kubernetes e2e suite}": "blah",
 	}
 	for input, output := range tests {
-		result := canonize(input)
+		result := normalize(input)
 		if result != output {
-			t.Errorf("canonize(%s) != %s (got %s)", input, output, result)
+			t.Errorf("normalize(%s) != %s (got %s)", input, output, result)
 		}
 	}
 }
 
 func TestOwnerList(t *testing.T) {
 	list := NewOwnerList(map[string]string{"Perf [performane]": "me"})
-	owner := list.Get("perf [flaky]")
+	owner := list.TestOwner("perf [flaky]")
 	if owner != "me" {
 		t.Error("Unexpected return value ", owner)
 	}
-	owner = list.Get("Unknown test")
+	owner = list.TestOwner("Unknown test")
 	if owner != "" {
 		t.Errorf("Unexpected return value ", owner)
 	}
 }
 
-func TestOwnerListFromTsv(t *testing.T) {
+func TestOwnerListFromCsv(t *testing.T) {
 	r := bytes.NewReader([]byte(",,header nonsense,\n" +
 		",owner,suggested owner,test name\n" +
 		",foo,other,Test name\n" +
 		",bar,foo,other test\n"))
-	list, err := NewOwnerListFromTsv(r)
+	list, err := NewOwnerListFromCsv(r)
 	if err != nil {
 		t.Error(err)
 	}
-	if owner := list.Get("test name"); owner != "foo" {
+	if owner := list.TestOwner("test name"); owner != "foo" {
 		t.Error("unexpected return value ", owner)
 	}
-	if owner := list.Get("other test"); owner != "bar" {
+	if owner := list.TestOwner("other test"); owner != "bar" {
 		t.Error("unexpected return value ", owner)
 	}
 }
+
+func TestReloadingOwnerList(t *testing.T) {
+	tempfile, err := ioutil.TempFile(os.TempDir(), "ownertest")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(tempfile.Name())
+	defer tempfile.Close()
+	writer := bufio.NewWriter(tempfile)
+	_, err = writer.WriteString("owner,test name\nfoo,flake\n")
+	if err != nil {
+		t.Error(err)
+	}
+	err = writer.Flush()
+	if err != nil {
+		t.Error(err)
+	}
+	list, err := NewReloadingOwnerList(tempfile.Name())
+	if err != nil {
+		t.Error(err)
+	}
+	if owner := list.TestOwner("flake"); owner != "foo" {
+		t.Error("unexpected owner for 'flake': ", owner)
+	}
+
+	// Assuming millisecond resolution on our FS, this sleep
+	// ensures the mtime will change with the next write.
+	time.Sleep(5 * time.Millisecond)
```

Comment:
I won't ask for a change, but if this is flaky, I'll be filing a bug against you to make a time checking interface instead of involving the filesystem in your test :)